clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -4259,7 +4240,7 @@ LValue CodeGenFunction::EmitArraySubscriptExpr(const ArraySubscriptExpr *E,
       if (const auto *ME = dyn_cast<MemberExpr>(Array);
           ME &&
           ME->isFlexibleArrayMemberLike(getContext(), StrictFlexArraysLevel) &&
-          ME->getMemberDecl()->hasAttr<CountedByAttr>()) {
+          ME->getMemberDecl()->getType()->getAs<CountAttributedType>()) {
```

Comment:
```suggestion
          isa<CountAttributedType>(ME->getMemberDecl()->getType()) {
```

Reply:
`ME->getMemberDecl()->getType()` might be wrapped in another sugar type, so `isa` might not work in that case. To deal with that, I just added `isCountAttributedType()`.